src/v8/inspector/mod.rs

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright Redis Ltd. 2022 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+//! The [Inspector] is a helpful facility for remote debugging of the
+//! JavaScript code provided by the V8 engine.
+//!
+//! To be able to debug, an inspector must be created and supplied with
+//! the remote debugger's requests. To answer back, the `on_response`
+//! callback must be properly set.
+//!
+//! # Sessions
+//!
+//! To be able to debug remotely, a server is required. Here one may
+//! find the [server::WebSocketServer] useful. The
+//! [server::WebSocketServer] is a very simple WebSocket server, which
+//! can be used with an [Inspector].
+//!
+//! To start a debugging server session, it is required to have a
+//! [V8ContextScope], a [server::WebSocketServer] and an [Inspector].
+//! All of those are gently packed within an easy-to-use
+//! [server::DebuggerSession] struct which abstracts all the work. It is
+//! a scoped object, meaning that once an object of
+//! [server::DebuggerSession] leaves its scope, the debugger server and
+//! the debugging session are stopped.
+//!
+//! In case the `"debug-server"` feature isn't enabled, the user of the
+//! crate must manually provide a way to receive and send messages over
+//! the network and feed the [Inspector] with data.
+use std::{ops::Deref, ptr::NonNull, sync::Arc};
+
+#[cfg(feature = "debug-server")]
+pub mod messages;
+#[cfg(feature = "debug-server")]
+pub mod server;
+
+use crate::v8_c_raw::bindings::v8_context_ref;
+
+use super::{isolate::V8Isolate, v8_context_scope::V8ContextScope};
+
+/// The debugging inspector, carefully wrapping the
+/// [`v8_inspector::Inspector`](https://chromium.googlesource.com/v8/v8/+/refs/heads/main/src/inspector)
+/// API. An inspector is tied to the [V8Isolate] it was created for.
+///
+/// # Example
+///
+/// ```rust
+/// use v8_rs::v8::*;
+/// use v8_rs::v8::inspector::RawInspector;
+///
+/// // Initialise the V8 engine:
+/// v8_init_platform(1, Some("--expose-gc")).unwrap();
+/// v8_init().unwrap();
+///
+/// // Create a new isolate:
+/// let isolate = isolate::V8Isolate::new();
+///
+/// // Enter the isolate created:
+/// let i_scope = isolate.enter();
+///
+/// // Create a JS execution context for code invocation:""
+/// let ctx = i_scope.new_context(None);
+///
+/// // Enter the created execution context for debugging:
+/// let ctx_scope = ctx.enter(&i_scope);
+///
+/// // Create an inspector.
+/// let _inspector = RawInspector::new(&ctx_scope);
```

Comment:
Why do we create the inspector from the ctx and not from the isolate? Is this inspector only relevant for this context?

Reply:
Can you be more specific where is it require? Could not see it.

Reply:
Yes but as far as I can see we only use it to get the isolate, so why not just pass isolate from the first place?

Reply:
Yes, just saw, thanks.